lxd/instance/instance_utils.go

```diff
@@ -658,7 +658,11 @@ func ValidName(instanceName string, isSnapshot bool) error {
 			return fmt.Errorf("Invalid instance name %q: %w", parentName, err)
 		}
 
-		// Snapshot part is more flexible, but doesn't allow space or / character.
+		// Snapshot part is more flexible, but doesn't allow "..", space or / characters.
```

Comment:
Should we also reject `*`?

```bash
$ lxc snapshot c1 '.*'
Error: Create instance snapshot: Failed to run: zfs snapshot -r default/containers/c1@snapshot-.*: exit status 1 (cannot create snapshot 'default/containers/c1@snapshot-.*': invalid character '*' in name
no snapshots were created)
```

And also `@` and `#` which have special meaning in ZFS at least.

Reply:
yeah maybe - although this might break users who've used snapshots with those chars in on non-zfs drivers.